### PR TITLE
Update links, edit/formatting pass for C++ IntelliSense

### PR DESCRIPTION
**Pull request by gregvanl**
Follow up PR for https://github.com/microsoft/vscode-docs/pull/6614
Also use Markdown `<details>` flyouts for sample sections

### CODE REVIEW

**Review by rzhao271 — APPROVED** [review]
So much better 🙏

File: docs/cpp/configure-intellisense.md [hunk 1]
```diff
@@ -4,63 +4,67 @@ Area: cpp
...
 
-If you do not see a status bar indicator, you can also check your project's `c_cpp_properties.json` file. This file stores all of your IntelliSense configuration settings. Navigate to this file by selecting **C/C++: Edit Configurations (UI)** from the Command Palette (`kb(workbench.action.showCommands)`). Check the `IntelliSense mode` to find your configuration. To learn more about the `c_cpp_properties.json` file, see the [schema reference](/docs/cpp/c-cpp-properties-schema-reference.md).
+If you do not see a status bar indicator, you can also check your project's `c_cpp_properties.json` file. This file stores all of your IntelliSense configuration settings. Navigate to this file by selecting **C/C++: Edit Configurations (UI)** from the Command Palette (`kb(workbench.action.showCommands)`). Check the `IntelliSense mode` to find your configuration. To learn more about the `c_cpp_properties.json` file, you can review the [schema reference](/docs/cpp/c-cpp-properties-schema-reference.md).
 
-![Command Palette](images/cpp/command-palette.png)
+![Command Palette filtered on C/C++ extension commands](images/cpp/command-palette.png)
```
Comment (AlexandraKemperMS) [inline docs/cpp/configure-intellisense.md:30]: When looking at the staged version, we can actually make this image a lot shorter, could we use this instead? 
image

File: docs/cpp/configure-intellisense.md [hunk 2]
```diff
@@ -4,63 +4,67 @@ Area: cpp
...
 IntelliSense configuration is stored in the `c_cpp_properties.json` file, which is automatically created in your workspace. All three of the following options are different ways of editing the `c_cpp_properties.json` file:
 
-### Option 1. Select a configuration option through the configuration quick pick
+### Option 1. Select a configuration option through the configuration Quick Pick
 
-Open the quick pick by entering **Select IntelliSense Configuration** in the Command Palette (`kb(workbench.action.showCommands)`), which shows you a dropdown with all of the configuration options found by the C/C++ Extension on your machine.
+Open the Quick Pick by entering **Select IntelliSense Configuration** in the Command Palette (`kb(workbench.action.showCommands)`), which shows you a dropdown with all of the configuration options found by the C/C++ extension on your machine.
 
-![Compiler Quick Pick](images/intellisense/compiler-quick-pick.png)
+![Select a compiler to configure for IntelliSense Quick Pick](images/intellisense/compiler-quick-pick.png)
```
Comment (AlexandraKemperMS) [inline docs/cpp/configure-intellisense.md:40]: Same here, we can make this shorter with this image: 
image

File: docs/cpp/configure-intellisense.md [hunk 3]
```diff
@@ -4,63 +4,67 @@ Area: cpp
...
 
 ### Option 2. Edit your IntelliSense configurations through the UI
 
-Open your IntelliSense configuration by selecting **C/C++: Edit Configurations (UI)** from the Command Palette (`kb(workbench.action.showCommands)`). This view is an interface of the `c_cpp_properties.json` file.
+Open your IntelliSense configuration by selecting **C/C++: Edit Configurations (UI)** from the Command Palette (`kb(workbench.action.showCommands)`). This view is a user interface for the `c_cpp_properties.json` file.
 
 Set the **Compiler path** field to the full path of the compiler you're using to build your project. For example, when using the default install path for GCC on Linux, the compiler path is `/usr/bin/gcc`. Set the **IntelliSense mode** to the architecture-specific variant of the compiler you're using.
```
Comment (AlexandraKemperMS) [inline docs/cpp/configure-intellisense.md:51]: There should also be a screenshot here: 
image

File: docs/cpp/configure-intellisense.md [hunk 4]
```diff
@@ -4,63 +4,67 @@ Area: cpp
...
 
-### Option 3. Edit the `c_cpp_properties.json` file directly
+### Option 3. Edit the c_cpp_properties.json file directly
 
 You can edit the `c_cpp_properties.json` file directly to customize your configuration. Use the **C/C++ Edit Configurations (JSON)** command from the Command Palette (`kb(workbench.action.showCommands)`), then the `c_cpp_properties.json` file is created in the `.vscode` folder of your workspace.
+
```
Comment (AlexandraKemperMS) [inline docs/cpp/configure-intellisense.md:55]: One more time: 
image

File: docs/cpp/configure-intellisense.md [hunk 5]
```diff
@@ -144,49 +150,52 @@ Select a tab to see some samples of a `c_cpp_configuration.json` file based on y
...
+A custom configuration provider is another extension in VS Code that can potentially provide more accurate C++ IntelliSense configuration than the C/C++ extension. For example, for the CMake or Make build systems, where the [Makefile Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools) or [CMake Tools](/docs/cpp/cmake-linux.md) extensions can be configuration providers. To add an extension as a configuration provider, either select the extension through the configuration Quick Pick, add it to configuration UI by editing the **Configuration provider** field under **Advanced Settings**, or add the `configurationProvider` field to your `c_cpp_properties.json` file. For example, for the CMake extension, the path to add would be `ms-vscode.cmake-tools`.
```
Comment (AlexandraKemperMS) [inline docs/cpp/configure-intellisense.md:172]: Please add this image: 
image

**Pull request merged**